lib/Optimizer/Transforms/GenKernelExecution.cpp

```diff
@@ -941,7 +950,8 @@ class GenerateKernelExecution
         cudaq::cc::numberOfHiddenArgs(hasThisPointer, hiddenSRet);
     if (count > 0 && args.size() >= count &&
         std::all_of(args.begin(), args.begin() + count, [](auto i) {
-          return isa<cudaq::cc::PointerType>(i.getType());
+          return isa<cudaq::cc::PointerType>(i.getType()) &&
+                 !isStatePointerType(i.getType());
```

Comment:
I don't get this one. We're dropping hidden arguments. Why would a `cudaq::state*` appear as a `this` or `sret`? Do we have an example?